rcl/src/rcl/arguments.c

```diff
@@ -1593,21 +1651,166 @@ _rcl_parse_remap_begin_remap_rule(
+
+  ret = rcl_lexer_lookahead2_init(&lex_lookahead, arg, *allocator);
+  if (RCL_RET_OK != ret) {
+    return ret;
   }
-  RCL_SET_ERROR_MSG("Argument does not use a valid severity level");
-  return RCL_RET_ERROR;
+
+  ret = _rcl_parse_log_level_name(&lex_lookahead, allocator, &logger_name);
+  if (RCL_RET_OK == ret) {
+    if (strlen(logger_name) == 0) {
+      RCL_SET_ERROR_MSG("Argument has an invalid logger item that name is empty");
+      ret = RCL_RET_INVALID_LOG_LEVEL_RULE;
+      goto cleanup;
+    }
+
+    ret = rcl_lexer_lookahead2_expect(&lex_lookahead, RCL_LEXEME_SEPARATOR, NULL, NULL);
+    if (RCL_RET_WRONG_LEXEME == ret) {
+      ret = RCL_RET_INVALID_LOG_LEVEL_RULE;
+      goto cleanup;
+    }
+
+    const char * level_token;
+    size_t level_token_length;
+    ret = rcl_lexer_lookahead2_expect(
+      &lex_lookahead, RCL_LEXEME_TOKEN, &level_token, &level_token_length);
+    if (RCL_RET_WRONG_LEXEME == ret) {
+      ret = RCL_RET_INVALID_LOG_LEVEL_RULE;
+      goto cleanup;
+    }
+
+    ret = rcl_lexer_lookahead2_expect(&lex_lookahead, RCL_LEXEME_EOF, NULL, NULL);
+    if (RCL_RET_OK != ret) {
+      ret = RCL_RET_INVALID_LOG_LEVEL_RULE;
+      goto cleanup;
+    }
+
+    logger_level = rcutils_strndup(level_token, level_token_length, *allocator);
+    if (NULL == logger_level) {
+      RCL_SET_ERROR_MSG("failed to copy logger level");
+      ret = RCL_RET_BAD_ALLOC;
+      goto cleanup;
+    }
+    rcutils_ret = rcutils_logging_severity_level_from_string(
+      logger_level, *allocator, &level);
+    if (RCUTILS_RET_OK == rcutils_ret) {
+      ret = rcl_log_levels_add_logger_setting(
+        log_levels, logger_name, (rcl_log_severity_t)level);
+      if (ret != RCL_RET_OK) {
+        goto cleanup;
+      }
+    }
+  } else {
+    rcutils_ret = rcutils_logging_severity_level_from_string(
+      arg, *allocator, &level);
+    if (RCUTILS_RET_OK == rcutils_ret) {
+      if (log_levels->default_logger_level != RCUTILS_LOG_SEVERITY_UNSET &&
+        log_levels->default_logger_level != (rcl_log_severity_t)level)
+      {
+        RCUTILS_LOG_DEBUG_NAMED(
+          ROS_PACKAGE_NAME, "Minimum default log level will be replaced from %d to %d",
+          log_levels->default_logger_level, level);
+      }
+      log_levels->default_logger_level = (rcl_log_severity_t)level;
+      ret = RCL_RET_OK;
+    }
+  }
+
+  if (RCUTILS_RET_OK != rcutils_ret) {
+    RCL_SET_ERROR_MSG("Argument does not use a valid severity level");
+    ret = RCL_RET_ERROR;
+  }
+
+cleanup:
+  if (logger_name) {
+    allocator->deallocate(logger_name, allocator->state);
+  }
+  if (logger_level) {
+    allocator->deallocate(logger_level, allocator->state);
+  }
+  if (RCL_RET_OK != ret) {
+    if (RCL_RET_OK != rcl_lexer_lookahead2_fini(&lex_lookahead)) {
+      RCUTILS_LOG_ERROR_NAMED(ROS_PACKAGE_NAME, "Failed to fini lookahead2 after error occurred");
+    }
+  } else {
+    ret = rcl_lexer_lookahead2_fini(&lex_lookahead);
+  }
```

Comment:
I find this a bit more clear:
```suggestion
  rcl_ret_t rv = rcl_lexer_lookahead2_fini(&lex_lookahead);
  if (RCL_RET_OK != rv) {
    if (RCL_RET_OK != ret) {
      RCUTILS_LOG_ERROR_NAMED(ROS_PACKAGE_NAME, "Failed to fini lookahead2 after error occurred");
    } else {
      ret = rv;
    }
  }
```
not a big difference though

Reply:
Updated